File: unit-3/sprint-3/day-3/we/BoundService/app/src/main/java/com/example/boundservice/MusicService.java
package com.example.boundservice;

import android.app.Service;
import android.content.Intent;
import android.media.MediaPlayer;
import android.os.Binder;
import android.os.IBinder;

public class MusicService extends Service {
    public MusicService() {
    }
    private MediaPlayer mediaPlayer;

    @Override
    public void onCreate() {
        super.onCreate();

    }

    @Override
    public IBinder onBind(Intent intent) {
        return new ServiceBinder();
    }

    public void play(){
        mediaPlayer = MediaPlayer.create(this,R.raw.music);
        if (!mediaPlayer.isPlaying()) mediaPlayer.start();
    }

    public void pause(){
        if (mediaPlayer.isPlaying()) mediaPlayer.pause();
    }

    public void stop(){
        if (mediaPlayer.isPlaying()) mediaPlayer.stop();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mediaPlayer.release();
    }

    public class ServiceBinder extends Binder {

        public MusicService getMusicService() {
            return MusicService.this;
        }

    }
}

File: unit-3/sprint-4/evaluation/App1/app/src/main/java/com/example/app1/MusicService.kt
package com.example.app1

import android.app.Service
import android.content.Intent
import android.os.IBinder

class MusicService : Service() {

    override fun onBind(intent: Intent): IBinder {
        TODO("Return the communication channel to the service.")
    }

    override fun onCreate() {
        super.onCreate()

    }

    override fun onDestroy() {
        super.onDestroy()

    }


}